broker-daemon/block-order-worker/index.spec.js

```diff
@@ -885,14 +794,140 @@ describe('BlockOrderWorker', () => {
+      })
+
+      it('gets a block order by id', async () => {
+        await worker.checkBlockOrderCompletion(blockOrder.id).catch(() => {})
+        expect(getBlockOrderStub).to.have.been.calledWith(blockOrder.id)
+      })
+    })
+
+    context('block order with openOrders', () => {
+      let blockOrder
+      let getBlockOrderStub
+      let worker
+      let orderStateMachines
+      let blockOrderCompleteStub
+
+      beforeEach(() => {
+        getBlockOrderStub = sinon.stub()
+        blockOrderCompleteStub = sinon.stub()
+        orderStateMachines = [
+          { order: { fillAmount: 100 } },
+          { order: { fillAmount: 200 } },
+          { order: { fillAmount: 300 } }
+        ]
+        blockOrder = {
+          id: '1234',
+          baseAmount: 600,
+          fills: [],
+          openOrders: orderStateMachines,
+          complete: blockOrderCompleteStub
+        }
+
+        worker = new BlockOrderWorker({ orderbooks, store, logger, relayer, engines })
+        worker.blockOrder = blockOrder
+        worker.getBlockOrder = getBlockOrderStub.returns(blockOrder)
+      })
+
+      it('completes a block order', async () => {
+        await worker.checkBlockOrderCompletion(blockOrder.id)
+        expect(blockOrderCompleteStub).to.have.been.calledOnce()
+      })
+
+      it('does not complete a block order if order is not filled', async () => {
+        const activeBlockOrder = {
+          id: '1234',
+          baseAmount: 600,
+          fills: [],
+          openOrders: [
+            { order: { fillAmount: 100 } }
+          ]
+        }
+        getBlockOrderStub.returns(activeBlockOrder)
+        await worker.checkBlockOrderCompletion(blockOrder.id)
+        expect(blockOrderCompleteStub).to.not.have.been.calledOnce()
+      })
+    })
+
+    context('block order with fills', () => {
+      let blockOrder
+      let getBlockOrderStub
+      let worker
+      let fillStateMachines
+      let blockOrderCompleteStub
+
+      beforeEach(() => {
+        getBlockOrderStub = sinon.stub()
+        blockOrderCompleteStub = sinon.stub()
+        fillStateMachines = [
+          { order: { fillAmount: 100 } },
+          { order: { fillAmount: 200 } },
+          { order: { fillAmount: 300 } }
+        ]
+        blockOrder = {
+          id: '1234',
+          baseAmount: 600,
+          fills: [],
+          openOrders: fillStateMachines,
+          complete: blockOrderCompleteStub
+        }
+
+        worker = new BlockOrderWorker({ orderbooks, store, logger, relayer, engines })
+        worker.blockOrder = blockOrder
+        worker.getBlockOrder = getBlockOrderStub.returns(blockOrder)
+      })
+
+      it('completes a block order', async () => {
+        await worker.checkBlockOrderCompletion(blockOrder.id)
+        expect(blockOrderCompleteStub).to.have.been.calledOnce()
+      })
+
+      it('does not complete a block order if order is still active', async () => {
```

Comment:
is this test mis-named? the underlying orders being active is no longer a condition